activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -842,7 +842,19 @@ def connection_pool_list
       def establish_connection(owner, spec)
         @class_to_pool.clear
         raise RuntimeError, "Anonymous class is not allowed." unless owner.name
-        owner_to_pool[owner.name] = ConnectionAdapters::ConnectionPool.new(spec)
+
+        message_bus = ActiveSupport::Notifications.instrumenter
+        payload = {
+          connection_id: object_id
+        }
+        payload[:class_name] = owner.name if owner
+        payload[:config] = spec.config if spec
+
+        message_bus.instrument('connection.active_record', payload) do
```

Comment:
Can we add a `!` prefix in the message to say people that it should not be used in production?